packages/block-editor/src/components/link-control/index.js

```diff
@@ -143,6 +143,11 @@ function LinkControl( {
 		return ( maybeURL ) ? handleDirectEntry( val, args ) : handleEntitySearch( val, args );
 	}, [ handleDirectEntry, fetchSearchSuggestions ] );
 
+	const forceUpdateInputValue = ( val ) => {
+		setInputValue( '' );
+		setInputValue( val );
```

Comment:
Interesting! Why is it necessary to double-call the setter with different values?

Reply:
This actually doesnt work. Looks like the wrong results appear. 

It "worked" because it forced the input to detect a change event which i assume doesnt otherwise happen because the value remains the same within the input HTML.

Reply:
I tried something similar with no luck. I thought about putting the reset in a timeout, but that felt even more hacky than it already was 😅

Reply:
yeah, this is definitely case of both changes being done in a single render so there won't be any update to DOM whatsoever. `setTimeout` to put it on the next frame will work but agreed it doesn't look great. Maybe there is some hook magic we could use? I'm not sure

Reply:
> both changes being done in a single render

Yeh, that's what I thought would happen. Then I tested it and I saw the results come up and thought...oh ok it works. But actually I was fooled! Going to revert this change.